docs/non-normative/how-to-write-conventions/resource-and-entities.md

```diff
@@ -0,0 +1,301 @@
+However, in practice, this is difficult or "non standard" in the following
+scenarios:
+
+- Prometheus pull metrics that want the `instance` label to match
+  `service.instance.id` on push based OTLP data.
+- Reading container logs from a `k8s.node`, where we know the container name
+  and deployment, but can't see into the SDK to understand a chosen instance id.
+
+The OpenTelemetry Operator, and onboarding guides for kubernetes, e.g. leverage
+mechanisms to ensure a `service.instance.id` can be pushed down to SDKs and
+external observers, alleviating this friction for kubernetes.
+
+### How to namespace entities?
+
+Entities should be namespaced around the primary mechanism used to
+identify the Entity. For Example, kubernetes entities use the `k8s`
+namespace, and are primarily discovered using the kubernetes API or
+working within kubernetes.
+
+See [General Naming Guidance](/docs/general/naming.md) for overall
+semantic convention namespacing rules.
+
+## Background: Resource and Entities
+
+In OpenTelemetry, every signal is associated with a Resource.
+According to the [Specification](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.45.0/specification/resource#overview)
+this is:
+
+> A Resource is a representation of the entity producing telemetry. Within
+> OpenTelemetry, all signals are associated with a Resource, enabling contextual
+> correlation of data from the same source. For example, if I see a high latency
+> in a span I need to check the metrics for the same entity that produced that
+> Span during the time when the latency was observed.
+>
+> Resource provides two important aspects for observability:
+>
+> * It MUST identify an entity that is producing telemetry.
+> * It SHOULD allow users to determine where that entity resides within their
+>   infrastructure.
+
+All resources are composed of [Entities](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.45.0/specification/entities/README.md#overview).
+An entity is specified as:
+
+> Entity represents an object of interest associated with
+> produced telemetry: traces, metrics, logs, profiles etc.
+
+While there is overlap in the definition of Entity and Resource, there are
+several key differences between the two:
+
+- An Entity has a known "type", e.g. `service`, `k8s.pod`,
+  `host`, etc.
+- An Entity can distinguish *identifying* attributes from
+  *descriptive* attributes.
+  - Identifying attributes can be used to identify the entity
+    within some system (See
+    [minimally sufficient id](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.45.0/specification/entities/data-model.md#minimally-sufficient-identity)).
+    For Example, the `k8s.pod.uid` would be considered an
+    identifying attribute for a pod within kubernetes.
+  - *Descriptive* attributes can be used to provide additional labels for
+    entities, but do not uniquely identify that Entity.  
```

Comment:
Nit: descriptive attributes may happen to uniquely identify an Entity. A better phrasing may be "are not necessary to uniquely identify".

Reply:
Ah right, for when we have options.  I'll update this.